File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-6.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l=0;
        int sum =0;
        int size=INT_MAX;
        for(int r=0; r<nums.size(); r++){
            sum+=nums[r];
            while(sum>=target){
                size = min(r-l+1, size);
                sum-=nums[l];
                l++;
                
            }
        }
        return size==INT_MAX ? 0:size;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-6.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        subset = []
        def dfs(i, sub):
            s = sum(sub)
            if s==target:
                res.append(sub.copy())
                return
            if s>target or i>=len(nums):
                return
            subset.append(nums[i])
            dfs(i, subset)
            subset.pop()
            dfs(i+1, subset)
        dfs(0,[])
        return res
            